pkg/convert/testdata/programs/expressions/pcl/main.pp

```diff
@@ -66,9 +66,9 @@
   value = {
     aTuple = ["a", "b", "c"]
     anObject = {
-      literalKey                = 1
-      anotherLiteralKey         = 2
-      "yet_another_literal_key" = aValue
+      literalKey             = 1
+      anotherLiteralKey      = 2
+      "yetAnotherLiteralKey" = aValue
```

Comment:
This change doesn't look right, a literal key should probably stay as is 🤔 we only rewrite object keys such as these when we know we can do that: for example terraform keys of snake_case matching camelCase pulumi fields. 

We would need to apply this rewrite whenever the program references it somewhere down the line:
```tf
locals {
  obj = {
    "snake_case_key_becomes_camel" = "abc123",
  }
}

output "referenced" {
  // is the referenced key rewritten here too? maybe we shouldn't actually rewrite these
  value = local.obj["snake_case_key_becomes_camel"]
}
```

Reply:
This is exactly the concern I had when writing it (I keep forgetting to mark things as draft when it is just an RFC).

The solution then would be to detect whenever we have a literal in a map and save it in a list of things NOT to mangle and then be sure to check it.  WDYT?

Reply:
I don't know about keeping track of things, instead _knowing_ when we should actually mangle or not and tend not to mangle when possible. Only mangle reference names, rewrite to match pulumi fields, otherwise keep it as is 🤔 ?

Reply:
yeah I think that will work